### ISSUE TITLE
Error with range and degrees

### ISSUE THREAD

**Issue by briochemc** [PainterQubits/Unitful.jl#241]
I'm not sure what's happening but here is a MWE throwing an `ERROR`:
```julia
julia> using Unitful

julia> x = range(0, 2, length=5) * u"°"
0.0°:0.5°:2.0°

julia> x[2:end]
Error showing value of type StepRangeLen{Base.TwicePrecision{Float64},Base.TwicePrecision{Float64},Base.TwicePrecision{Quantity{Float64,NoDims,Unitful.FreeUnits{(°,),NoDims,nothing}}}}:
ERROR: MethodError: no method matching Float64(::Quantity{Float64,NoDims,Unitful.FreeUnits{(°,),NoDims,nothing}})
Closest candidates are:
  Float64(::T<:Number) where T<:Number at boot.jl:741
  Float64(::Int8) at float.jl:60
  Float64(::Int16) at float.jl:60
  ...
Stacktrace:
 [1] Float64(::Base.TwicePrecision{Quantity{Float64,NoDims,Unitful.FreeUnits{(°,),NoDims,nothing}}}) at ./twiceprecision.jl:243
 [2] convert(::Type{Float64}, ::Base.TwicePrecision{Quantity{Float64,NoDims,Unitful.FreeUnits{(°,),NoDims,nothing}}}) at ./twiceprecision.jl:250
 [3] Base.TwicePrecision{Float64}(::Base.TwicePrecision{Quantity{Float64,NoDims,Unitful.FreeUnits{(°,),NoDims,nothing}}}) at ./twiceprecision.jl:192
 [4] step(::StepRangeLen{Base.TwicePrecision{Float64},Base.TwicePrecision{Float64},Base.TwicePrecision{Quantity{Float64,NoDims,Unitful.FreeUnits{(°,),NoDims,nothing}}}}) at ./range.jl:500
 [5] show(::IOContext{REPL.Terminals.TTYTerminal}, ::StepRangeLen{Base.TwicePrecision{Float64},Base.TwicePrecision{Float64},Base.TwicePrecision{Quantity{Float64,NoDims,Unitful.FreeUnits{(°,),NoDims,nothing}}}}) at ./range.jl:711
 [6] show(::IOContext{REPL.Terminals.TTYTerminal}, ::MIME{Symbol("text/plain")}, ::StepRangeLen{Base.TwicePrecision{Float64},Base.TwicePrecision{Float64},Base.TwicePrecision{Quantity{Float64,NoDims,Unitful.FreeUnits{(°,),NoDims,nothing}}}}) at ./show.jl:7
 [7] display(::REPL.REPLDisplay{REPL.LineEditREPL}, ::MIME{Symbol("text/plain")}, ::StepRangeLen{Base.TwicePrecision{Float64},Base.TwicePrecision{Float64},Base.TwicePrecision{Quantity{Float64,NoDims,Unitful.FreeUnits{(°,),NoDims,nothing}}}}) at /Users/benoitpasquier/.julia/packages/OhMyREPL/GFHgr/src/output_prompt_overwrite.jl:6
 [8] display(::REPL.REPLDisplay, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:135
 [9] display(::Any) at ./multimedia.jl:287
 [10] #invokelatest#1 at ./essentials.jl:742 [inlined]
 [11] invokelatest at ./essentials.jl:741 [inlined]
 [12] print_response(::IO, ::Any, ::Any, ::Bool, ::Bool, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:155
 [13] print_response(::REPL.AbstractREPL, ::Any, ::Any, ::Bool, ::Bool) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:140
 [14] (::getfield(REPL, Symbol("#do_respond#38")){Bool,getfield(REPL, Symbol("##48#57")){REPL.LineEditREPL},REPL.LineEditREPL,REPL.LineEdit.Prompt})(::Any, ::Any, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:714
 [15] #invokelatest#1 at ./essentials.jl:742 [inlined]
 [16] invokelatest at ./essentials.jl:741 [inlined]
 [17] run_interface(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/REPL/src/LineEdit.jl:2273
 [18] run_frontend(::REPL.LineEditREPL, ::REPL.REPLBackendRef) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:1035
 [19] run_repl(::REPL.AbstractREPL, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:192
 [20] (::getfield(Base, Symbol("##734#736")){Bool,Bool,Bool,Bool})(::Module) at ./client.jl:362
 [21] #invokelatest#1 at ./essentials.jl:742 [inlined]
 [22] invokelatest at ./essentials.jl:741 [inlined]
 [23] run_main_repl(::Bool, ::Bool, ::Bool, ::Bool, ::Bool) at ./client.jl:346
 [24] exec_options(::Base.JLOptions) at ./client.jl:284
 [25] _start() at ./client.jl:436
```
This does not seem to happen for different units, e.g.,
```julia
julia> x = range(0, 2, length=5) * u"m"
0.0 m:0.5 m:2.0 m

julia> x[2:end]
0.5 m:0.5 m:2.0 m
```

Any idea what's going on?

**Comment by timholy** [comment]
Here's the fundamental problem (discovered by stepping through `getindex(x, 2:5)` in the debugger):
```julia
julia> y = 1.0 * u"°"
1.0°

julia> y+y
2.0°

julia> y2 = Base.TwicePrecision(y)
Base.TwicePrecision{Quantity{Float64,NoDims,Unitful.FreeUnits{(°,),NoDims,nothing}}}(1.0°, 0.0°)

julia> y2 + y2
Base.TwicePrecision{Quantity{Float64,NoDims,Unitful.FreeUnits{(°,),NoDims,nothing}}}(2.0°, 0.0°)

julia> y2 + 2*y2
Base.TwicePrecision{Float64}(0.05235987755982989, -3.469446951953614e-18)

julia> 2*y2
Base.TwicePrecision{Float64}(0.03490658503988659, 0.0)
```

**Comment by briochemc** [comment]
So the problem is that degrees are converted to radians? I'm still confused though. Is the problem a Unitful one or a TwicePrecisison one?

**Comment by timholy** [comment]
Here's a snapshot of stepping into `2*y2` using Rebugger (deleting some extraneous lines...):
```julia
interpret> 2*y2
*(v::Number, x::Base.TwicePrecision) in Base at twiceprecision.jl:291
 *(x::Base.TwicePrecision{Q}, v::Real) where Q<:Quantity in Unitful at /home/tim/.julia/packages/Unitful/W0mMi/src/units.jl:281
  *(x::Base.TwicePrecision, y::Base.TwicePrecision) in Base at twiceprecision.jl:298
   promote(x, y) in Base at promotion.jl:283
  x = Base.TwicePrecision{Quantity{Float…its{(°,),NoDims,nothing}}}(1.0°, 0.0°)
  y = Base.TwicePrecision{Float64}(2.0, 0.0)
  px = Base.TwicePrecision{Float64}(0.017453292519943295, 0.0)
  #temp# = 2
  py = Base.TwicePrecision{Float64}(2.0, 0.0)
 283       @_inline_meta
 284      (px, py) = _promote(x, y)
 285      not_sametype((x, y), (px, py))
 286      (px, py)
      end
```
So the problem is that degrees get promoted to `Float64` by the Base routine. It might be better to rewrite https://github.com/PainterQubits/Unitful.jl/blob/45a5dff4afd7eea42a10865167e3201a376b13c8/src/units.jl#L282 to strip the units from `x`, call the Base TwicePrecision routines, and then re-add the units.

**Comment by ajkeller34** [comment]
Fixed by #245.
